### PR TITLE
[dunfell][gatesgarth][hardknott][master] Upgrade ROS 2 rolling to 2021-03-17 release

### PR DESCRIPTION
**Pull request by shr-project**
- Rolling 2020-12-29 - https://discourse.ros.org/t/new-packages-for-ros-2-rolling-ridley-2020-12-29/18233
- Rolling 2021-02-19 - https://discourse.ros.org/t/new-packages-for-ros-2-rolling-ridley-2021-02-19/19047
- Rolling 2021-03-17 - https://discourse.ros.org/t/new-packages-for-ros-2-rolling-ridley-2021-03-17/19523

### CODE REVIEW

**Comment by shr-project** [comment]
Remaining issues to resolve:
```
  meta-ros2-rolling/generated-recipes/rosidl-typesupport/rosidl-typesupport-cpp_1.1.1-1.bb:do_configure
  meta-ros2-rolling/generated-recipes/libyaml-vendor/libyaml-vendor_1.1.0-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/rosbag2/zstd-vendor_0.6.0-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/rosbag2/rosbag2-compression_0.6.0-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/rc-genicam-api/rc-genicam-api_2.4.4-1.bb:do_configure
  virtual:native:meta-ros2-rolling/generated-recipes/uncrustify-vendor/uncrustify-vendor_1.5.2-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/uncrustify-vendor/uncrustify-vendor_1.5.2-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/image-pipeline/image-publisher_2.2.1-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/rc-dynamics-api/rc-dynamics-api_0.10.3-1.bb:do_compile
  meta-ros2-rolling/generated-recipes/async-web-server-cpp/async-web-server-cpp_2.0.0-1.bb:do_package_qa
```

**Comment by shr-project** [comment]
Remaining issues after upgrade to 2021-03-17 release:
```
Summary: 10 tasks failed:
  meta-ros2-rolling/generated-recipes/rosidl-typesupport/rosidl-typesupport-cpp_1.1.1-3.bb:do_configure
  meta-ros2-rolling/generated-recipes/libyaml-vendor/libyaml-vendor_1.1.1-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/rosbag2/zstd-vendor_0.6.0-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/rosbag2/rosbag2-compression_0.6.0-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/rc-genicam-api/rc-genicam-api_2.5.0-1.bb:do_configure
  virtual:native:meta-ros2-rolling/generated-recipes/uncrustify-vendor/uncrustify-vendor_1.5.2-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/uncrustify-vendor/uncrustify-vendor_1.5.2-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/image-pipeline/image-publisher_2.2.1-1.bb:do_patch
  meta-ros2-rolling/generated-recipes/async-web-server-cpp/async-web-server-cpp_2.0.0-1.bb:do_package_qa
  meta-ros2-rolling/generated-recipes/rc-dynamics-api/rc-dynamics-api_0.10.3-1.bb:do_compile
Summary: There were 2 WARNING messages shown.
Summary: There were 13 ERROR messages shown, returning a non-zero exit code.
```

**Comment by shr-project** [comment]
Remaining issues after upgrade to 2021-03-17 release after fixing the issues reported above:
```
Summary: 11 tasks failed:
  /jenkins/mjansa/build/ros/webos-rolling-hardknott/meta-ros/meta-ros2-rolling/generated-recipes/rclcpp/rclcpp_6.3.1-1.bb:do_compile
  /jenkins/mjansa/build/ros/webos-rolling-hardknott/meta-ros/meta-ros2-rolling/generated-recipes/rcl/rcl-lifecycle_2.5.2-1.bb:do_compile
  /jenkins/mjansa/build/ros/webos-rolling-hardknott/meta-ros/meta-ros2-rolling/generated-recipes/bond-core/bond_3.0.1-2.bb:do_compile
  /jenkins/mjansa/build/ros/webos-rolling-hardknott/meta-ros/meta-ros2-rolling/generated-recipes/ublox/ublox-msgs_2.0.0-1.bb:do_configure
  /jenkins/mjansa/build/ros/webos-rolling-hardknott/meta-ros/meta-ros2-rolling/generated-recipes/grbl-msgs/grbl-msgs_0.0.2-1.bb:do_configure
  /jenkins/mjansa/build/ros/webos-rolling-hardknott/meta-ros/meta-ros2-rolling/generated-recipes/teleop-tools/teleop-tools-msgs_1.2.0-1.bb:do_compile
  /jenkins/mjansa/build/ros/webos-rolling-hardknott/meta-ros/meta-ros2-rolling/generated-recipes/rosbag2/rosbag2-py_0.6.0-1.bb:do_configure
  /jenkins/mjansa/build/ros/webos-rolling-hardknott/meta-ros/meta-ros2-rolling/generated-recipes/webots-ros2/webots-ros2-msgs_1.0.2-2.bb:do_compile
  /jenkins/mjansa/build/ros/webos-rolling-hardknott/meta-ros/meta-ros2-rolling/generated-recipes/udp-msgs/udp-msgs_0.0.3-1.bb:do_compile
  /jenkins/mjansa/build/ros/webos-rolling-hardknott/meta-ros/meta-ros2-rolling/generated-recipes/joystick-drivers/wiimote-msgs_3.0.0-2.bb:do_compile
  /jenkins/mjansa/build/ros/webos-rolling-hardknott/meta-ros/meta-ros2-rolling/generated-recipes/rc-common-msgs/rc-common-msgs_0.5.3-1.bb:do_compile
Summary: There were 11 ERROR messages shown, returning a non-zero exit code.
```

File: meta-ros2-rolling/recipes-bbappends/rosbag2/rosbag2-compression/0001-CMakeLists.txt-drop-dependency-on-zstd_vendor.patch [hunk 1]
```diff
@@ -38,12 +38,34 @@ index 176e71ff..e76ab41b 100644
...
  # This is appropriate when building the dll but not consuming it.
  target_compile_definitions(${PROJECT_NAME}_zstd
-@@ -86,7 +86,7 @@ ament_export_include_directories(include)
+@@ -86,7 +86,26 @@ ament_export_include_directories(include)
  ament_export_libraries(${PROJECT_NAME} ${PROJECT_NAME}_zstd)
  ament_export_targets(export_${PROJECT_NAME})
  # order matters here, first vendor, then zstd
 -ament_export_dependencies(pluginlib rcpputils rcutils rosbag2_cpp rosbag2_storage zstd_vendor zstd)
-+ament_export_dependencies(pluginlib rcpputils rcutils rosbag2_cpp rosbag2_storage zstd)
++# don't export zstd as we don't provide zstd from zstd-vendor and rosbag2-{py,transport} would fail with:
```
Comment (herb-kuta-lge) [inline meta-ros2-rolling/recipes-bbappends/rosbag2/rosbag2-compression/0001-CMakeLists.txt-drop-dependency-on-zstd_vendor.patch:46]: > don't export zstd

But I see `zstd` in `ament_export_dependencies()`. Perhaps you mean `zstd_vendor`?

File: meta-ros2-rolling/recipes-bbappends/grbl-msgs/grbl-msgs_0.0.2-1.bbappend [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2020 LG Electronics, Inc.
```
Comment (herb-kuta-lge) [inline meta-ros2-rolling/recipes-bbappends/grbl-msgs/grbl-msgs_0.0.2-1.bbappend:1]: 2020 -> 2021

File: meta-ros2-rolling/recipes-bbappends/rc-common-msgs/rc-common-msgs_0.5.3-1.bbappend [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2020 LG Electronics, Inc.
```
Comment (herb-kuta-lge) [inline meta-ros2-rolling/recipes-bbappends/rc-common-msgs/rc-common-msgs_0.5.3-1.bbappend:1]: 2020 -> 2021 (+ 3 more in new files).

**Review by herb-kuta-lge — CHANGES_REQUESTED** [review]

File: meta-ros2-rolling/recipes-bbappends/rosbag2/rosbag2-compression/0001-CMakeLists.txt-drop-dependency-on-zstd_vendor.patch [hunk 4]
```diff
@@ -38,12 +38,34 @@ index 176e71ff..e76ab41b 100644
...
++#     zstd-config.cmake
++#
++#   Add the installation prefix of "zstd" to CMAKE_PREFIX_PATH or set
++#   "zstd_DIR" to a directory containing one of the above files.  If "zstd"
++#   provides a separate development package or SDK, be sure it has been
++#   installed.
++# Call Stack (most recent call first):
++#   TOPDIR/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/rosbag2-py/0.6.0-1-r0/recipe-sysroot/usr/share/rosbag2_compression/cmake/rosbag2_compressionConfig.cmake:41 (include)
++#   CMakeLists.txt:21 (find_package)
++ament_export_dependencies(pluginlib rcpputils rcutils rosbag2_cpp rosbag2_storage)
```
Comment (shr-project) [inline meta-ros2-rolling/recipes-bbappends/rosbag2/rosbag2-compression/0001-CMakeLists.txt-drop-dependency-on-zstd_vendor.patch:65]: I don't see zstd here.

File: meta-ros2-rolling/recipes-bbappends/rosbag2/rosbag2-compression/0001-CMakeLists.txt-drop-dependency-on-zstd_vendor.patch (hunk 1 above)
Comment (shr-project) [reply meta-ros2-rolling/recipes-bbappends/rosbag2/rosbag2-compression/0001-CMakeLists.txt-drop-dependency-on-zstd_vendor.patch:46]: In the removed ament_export_dependencies() here, but not on the line 65 bellow this comment, right?

File: meta-ros2-rolling/recipes-bbappends/rc-common-msgs/rc-common-msgs_0.5.3-1.bbappend (hunk 3 above)
Comment (shr-project) [reply meta-ros2-rolling/recipes-bbappends/rc-common-msgs/rc-common-msgs_0.5.3-1.bbappend:1]: I wasn't updating it only the bbappends just imported from other layer.

File: meta-ros2-rolling/recipes-bbappends/grbl-msgs/grbl-msgs_0.0.2-1.bbappend (hunk 2 above)
Comment (shr-project) [reply meta-ros2-rolling/recipes-bbappends/grbl-msgs/grbl-msgs_0.0.2-1.bbappend:1]: I wasn't updating it only the bbappends just imported from other layer without any modification.

File: meta-ros2-rolling/recipes-bbappends/rosbag2/rosbag2-compression/0001-CMakeLists.txt-drop-dependency-on-zstd_vendor.patch (hunk 1 above)
Comment (herb-kuta-lge) [reply meta-ros2-rolling/recipes-bbappends/rosbag2/rosbag2-compression/0001-CMakeLists.txt-drop-dependency-on-zstd_vendor.patch:46]: Sorry, I was confused by GitHub's incorrect colorization of the change.

File: meta-ros2-rolling/recipes-bbappends/rosbag2/rosbag2-compression/0001-CMakeLists.txt-drop-dependency-on-zstd_vendor.patch (hunk 4 above)
Comment (herb-kuta-lge) [reply meta-ros2-rolling/recipes-bbappends/rosbag2/rosbag2-compression/0001-CMakeLists.txt-drop-dependency-on-zstd_vendor.patch:65]: Sorry, I was confused by GitHub's incorrect colorization of the change.

**Review by herb-kuta-lge — APPROVED** [review]

**Comment by shr-project** [comment]
Depends on https://github.com/ros/meta-ros/pull/824

**Comment by shr-project** [comment]
Pushed to other branches as well:
   920d6546c..8c93065d2  dunfell -> dunfell-next
   46eddc91f..a9a48d992  gatesgarth -> gatesgarth
   46eddc91f..a9a48d992  gatesgarth -> gatesgarth-next
   860df105d..89892e439  hardknott -> hardknott
   860df105d..89892e439  hardknott -> hardknott-next
   860df105d..89892e439  master -> master
   860df105d..89892e439  master -> master-next

**Pull request merged**